### PR TITLE
Get rid of "Improper Null Termination" issue

### PR DESCRIPTION
**Pull request by antonwolfy**
Add a null-terminated symbol at the end of char array to avoid "Improper Null Termination" issue reported by Checkmarx scan.

- [X] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [X] Have you tested your changes locally for CPU and GPU devices?
- [X] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

**Comment by oleksandr-pavlyk** [comment]
This change is not correct. Even though the pointer type is `char *` it is really meant as typeless pointer. The proper way to resolve the warning is to do `static_cast<void *>(char_ptr)` in the argument of `memcpy`.

The buffer is allocated of the size `itemsize` to store an item content, not a string.

**Comment by antonwolfy** [comment]
> This change is not correct. Even though the pointer type is `char *` it is really meant as typeless pointer. The proper way to resolve the warning is to do `static_cast<void *>(char_ptr)` in the argument of `memcpy`.
> 
> The buffer is allocated of the size `itemsize` to store an item content, not a string.

The solution is reworked in a bit different way than suggested. The buffer initially is declared as a pointer on void, since no pointer arithmetic required. Also the code around was a bit simplified with direct q.memcpy() call instead of submitting the kernel with single mempy() to queue.

File: dpnp/backend/kernels/dpnp_krnl_random.cpp [hunk 1]
```diff
@@ -1674,23 +1674,15 @@ DPCTLSyclEventRef dpnp_rng_shuffle_c(DPCTLSyclQueueRef q_ref,
         // Fast, statically typed path: shuffle the underlying buffer.
         // Only for non-empty, 1d objects of class ndarray (subclasses such
         // as MaskedArrays may not support this approach).
-        char* buf = reinterpret_cast<char*>(sycl::malloc_shared(itemsize * sizeof(char), q));
+        void* buf = sycl::malloc_device<void *>(itemsize, q);
```
Comment (oleksandr-pavlyk) [inline dpnp/backend/kernels/dpnp_krnl_random.cpp:1677]:
```suggestion
        void* buf = sycl::malloc_device(itemsize, q);
```

File: dpnp/backend/kernels/dpnp_krnl_random.cpp [hunk 2]
```diff
@@ -1699,23 +1691,15 @@ DPCTLSyclEventRef dpnp_rng_shuffle_c(DPCTLSyclQueueRef q_ref,
     {
         // Multidimensional ndarrays require a bounce buffer.
         size_t step_size = (size / high_dim_size) * itemsize; // size in bytes for x[i] element
-        char* buf = reinterpret_cast<char*>(sycl::malloc_shared(step_size * sizeof(char), q));
+        void* buf = sycl::malloc_device<void *>(step_size, q);
```
Comment (oleksandr-pavlyk) [inline dpnp/backend/kernels/dpnp_krnl_random.cpp:1694]:
```suggestion
        void* buf = sycl::malloc_device(step_size, q);
```

**Review by oleksandr-pavlyk — APPROVED** [review]

**Comment by antonwolfy** [comment]
@oleksandr-pavlyk , may I ask you to merge without passing the checks, no new issues observed there? (I personally don't have such permissions.)

**Pull request merged**